Continue with logging error message when log directory is already created

fsat opened the pull request:
Instead of failing with error complaing log directory is present.

fsat committed:
Continue with logging error message when log directory is already created

Instead of failing with error complaing log directory is present.

fsat commented:
Manual test is completed successfully.

**Setup**

Patch `conduct_version.py` to fail.

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ git diff
diff --git a/conductr_cli/conduct_version.py b/conductr_cli/conduct_version.py
index 79adf20..0a659e7 100644
--- a/conductr_cli/conduct_version.py
+++ b/conductr_cli/conduct_version.py
@@ -11,4 +11,5 @@ def version(args):
     log = logging.getLogger(__name__)
     log.screen(__version__)
     log.screen('Supported API version(s): {}'.format(', '.join(supported_api_versions())))
+    raise "boom"
     return True
```

Install the CLI locally using `pip3 install -e .`

Remove the `~/.conductr` directory.

**Result**

Run `conduct version` and inspect the log directory.

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ ll ~/.conductr
ls: /Users/felixsatyaputra/.conductr: No such file or directory

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ conduct version
0.41
Supported API version(s): 1, 2
Error: Encountered unexpected error.
Error: Reason: TypeError exceptions must derive from BaseException
Error: Further information of the error can be found in the error log file: /Users/felixsatyaputra/.conductr/errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ ll ~/.conductr
total 8
drwxr-xr-x   3 felixsatyaputra  staff   102B  4 Nov 10:53 .
drwxr-xr-x+ 66 felixsatyaputra  staff   2.2K  4 Nov 10:53 ..
-rw-r--r--   1 felixsatyaputra  staff   772B  4 Nov 10:53 errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ cat ~/.conductr/errors.log
2016-11-04 10:53:04,511: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
```

Re-run `conduct version again`, logs are appended as expected (i.e. no error in relation to log directory being present when calling `os.makedirs`)

```
Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ conduct version
0.41
Supported API version(s): 1, 2
Error: Encountered unexpected error.
Error: Reason: TypeError exceptions must derive from BaseException
Error: Further information of the error can be found in the error log file: /Users/felixsatyaputra/.conductr/errors.log

Felixs-MacBook-Pro-2:conductr-cli felixsatyaputra$ cat ~/.conductr/errors.log
2016-11-04 10:53:04,511: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
2016-11-04 10:54:12,885: Failure running the following command: ['/usr/local/bin/conduct', 'version']
Traceback (most recent call last):
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/main_handler.py", line 9, in run
    result = callback()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_main.py", line 371, in run
    is_completed_without_error = args.func(args)
  File "/Users/felixsatyaputra/workspace/typesafe-fsat/conductr-cli/conductr_cli/conduct_version.py", line 14, in version
    raise "boom"
TypeError: exceptions must derive from BaseException
```

huntc commented on conductr_cli/main_handler.py:
```diff
@@ -19,7 +19,7 @@ def run(callback):
         # Ensure log dir is present before errors are being logged
         log_dir = os.path.abspath(os.path.join(DEFAULT_ERROR_LOG_FILE, '..'))
-        os.makedirs(log_dir)
+        os.makedirs(log_dir, exist_ok=True)
```
Can you check for other instances of `makedirs` and make a similar change? I know we have one for the DC/OS integration...

fsat replied:
Ah yes - good idea. I'll do that.

fsat replied:
This is now done as separate commit: https://github.com/typesafehub/conductr-cli/pull/182/commits/a1ebd41e3f7864536c20074c8ad660e3e395cab3

huntc reviewed (changes requested):
One small request

fsat committed:
Ensure other makedirs have exist_ok=True specified

This is to ensure makedirs operation won't fail if the directory to be created is already present.

fsat commented:
![](https://github.trello.services/images/mini-trello-icon.png) [ConductR CLI: Fixes prior 2.0.0-RC1](https://trello.com/c/AXKwNWwh/379-conductr-cli-fixes-prior-2-0-0-rc1)
